docs/excel/excel-add-ins-pivots.md

```diff
@@ -0,0 +1,243 @@
+Manually created PivotTables are also accessible through the PivotTable collection of the workbook or of individual worksheets. 
+The following code gets the first PivotTable in the worksheet. It then gives the table a name for easy future reference.
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTableList = context.workbook.worksheets.getActiveWorksheet().pivotTables;
+	pivotTableList.load("no-properties-needed");
+	await context.sync();
+
+	const pivotTable = pivotTableList.items[0];
+	pivotTable.name = "My Pivot";
+	await context.sync();
+});
+```
+
+## Add rows and columns to a PivotTable
+
+Rows and columns pivot the data around those fields’ values.
+
+Adding the **Farm** column pivots all the sales around each farm. Adding the **Type** and **Classification** rows further breaks down the data based on what fruit was sold and whether it was organic or not.
+
+![A PivotTable with a Farm column and Type and Classification rows.](../images/excel-pivots-table-rows-and-columns.png)
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+
+	pivotTable.columnHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+
+	await context.sync();
+});
+```
+
+You can also have a PivotTable with only rows or columns.
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Classification"));
+	
+	await context.sync();
+});
+```
+
+## Add data hierarchies to the PivotTable
+
+Data hierarchies fill the PivotTable with information to combine based on the rows and columns. 
+Adding the data hierarchies of **Crates Sold at Farm** and **Crates Sold Wholesale** gives sums of those figures for each row and column. 
+In the example, both **Farm** and **Type** are rows, with the crate sales as the data. 
+
+![A PivotTable showing the total sales of different fruit based on the farm they came from.](../images/excel-pivots-data-hierarchy.png)
+
+```ts
+await Excel.run(async (context) => {
+	const pivotTable = context.workbook.worksheets.getActiveWorksheet().pivotTables.getItem("Farm Sales");
+
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Farm"));
+	pivotTable.rowHierarchies.add(pivotTable.hierarchies.getItem("Type"));
+
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold at Farm"));
+	pivotTable.dataHierarchies.add(pivotTable.hierarchies.getItem("Crates Sold Wholesale"));
+
+	await context.sync();
+});
+```
+
+## Change aggregation function
```

Comment:
lets add all the supported aggregation functions.(just mention and probably a sample that has requires a base field and a base item such as "Difference From "

Reply:
The baseField and baseItem stuff wasn't working previously (as per our email thread). I'll make a separate work item to go back and document those once the CDN is refreshed.